src/web/src/data/api.ts

```diff
@@ -0,0 +1,81 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { Breed, Favourite, Cat } from "../types";
+
+export default class Api {
+  client: AxiosInstance;
+  tokenId: string = "";
+
+  constructor(customClient?: AxiosInstance) {
+    if (customClient === undefined) {
+      const ssl = process.env.REACT_APP_BACKEND_SSL;
+      const host = process.env.REACT_APP_BACKEND_HOST;
+      const port = process.env.REACT_APP_BACKEND_PORT;
+      const isDemo = process.env.REACT_APP_DEMO_USER_ON;
+
+      var headers = {
+        "Content-Type": "application/json",
+        "token-id": isDemo === "true" ? "DEMO" : this.tokenId,
+      };
+
+      this.client = axios.create({
+        baseURL: `${ssl === "true" ? "https" : "http"}://${host}:${port}/api`,
+        headers,
+      });
+
+      this.client.interceptors.response.use(
+        (response) => response,
+        (error) => {
+          if (error.response.status === 401) {
+            window.location.href = "/auth/login";
+          }
+        }
+      );
+    } else {
+      this.client = customClient;
+    }
```

Comment:
If you inverted the if logic then you could have gotten away with nesting ;)

Reply:
True! :)